api/envoy/config/listener/v3/quic_config.proto

```diff
@@ -24,11 +24,17 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // [#protodoc-title: QUIC listener config]
 
 // Configuration specific to the UDP QUIC listener.
-// [#next-free-field: 12]
+// [#next-free-field: 13]
 message QuicProtocolOptions {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.api.v2.listener.QuicProtocolOptions";
 
+  message SaveCmsgConfig {
```

Comment:
Naming is one of the two hard problems in computer science (along with cache invalidation and off by one errors), but I'm not loving "SaveCmsgConfig". I'm also not finding obviously better names. I think that "Save" reads like a verb, so this seems like it's a function name instead of a data structure. Maybe "SavedCmsgConfig"? Oh, or perhaps "CmsReportingConfig" since it configures how CMSG is reported? That seems a bit cleaner to me. (or maybe CmsgSavingConfig if we think "save" is more correct than "report". Or maybe "processing" or "handling"?)

WDYT?

Reply:
Yup, I don't love the naming either, it was loosely inspired by [TCP_SAVE_SYN](https://lore.kernel.org/all/20150501201417.GB6113@akamai.com/T/).

I think I like "SavedCmsgConfig" best from your alternatives, but I don't love it. Maybe "CmsgToSaveConfig"? Let's settle on an alternative before I change all the callsites.

Reply:
Ah, interesting! I see how that maps to what you wrote. I'm happy to go with "SaveCmsgConfig" (even though I still don't love it :>). Thanks for the explanation.